src/main/java/run/halo/app/service/impl/CategoryServiceImpl.java

```diff
@@ -661,15 +662,21 @@ private boolean doCategoryHasEncrypt(
 
 
     @Override
+    @Transactional(rollbackFor = Exception.class)
     public List<Category> updateInBatch(Collection<Category> categories) {
         if (CollectionUtils.isEmpty(categories)) {
             return Collections.emptyList();
         }
-
-        ArrayList<Category> resultList = new ArrayList<>();
-        for (Category category : categories) {
-            resultList.add(update(category));
-        }
-        return resultList;
+        Set<Integer> categoryIds = ServiceUtils.fetchProperty(categories, Category::getId);
+        Map<Integer, Category> idCategoryParamMap =
+            ServiceUtils.convertToMap(categories, Category::getId);
+        return categoryRepository.findAllById(categoryIds)
+            .stream()
+            .map(categoryToUse -> {
```

Comment:
I suggesting naming `categoryToUse` to `categoryToUpdate`.

Reply:
ok